### PR TITLE
docs: updated Azure AD instructions for user management

### PR DESCRIPTION
**Pull request by schakko**
The current documentation for setting up the Azure AD user management integration contains duplicate content and misses some instructions.
Content of this PR:

* moved RBAC to its own section to prevent duplicate content
* added scenario for mapping security groups to RBAC roles
* added scenario for mapping application roles to RBAC roles
* fixed indention of `scope` YAML key

Signed-off-by: Christopher Klein <ckl@dreitier.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x]  (c) Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/user-management/microsoft.md [hunk 1]
```diff
@@ -92,107 +80,129 @@
...
-                     clientID: {azure_ad_application_client_id}
-                     clientSecret: $oidc.azure.clientSecret
-                     requestedIDTokenClaims:
-                        groups:
-                           essential: true
-                     requestedScopes:
-                        - openid
-                        - profile
-                        - email
+```yaml
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:109]: Unfortunately, the absurd-looking indentation is necessary to maintain list ordering when rendered by mkdocs. You can test with `make server-docs-local1.

image

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks @schakko! One top-level commend, then I'll come back for a more thorough review.  :-)

File: docs/operator-manual/user-management/microsoft.md (hunk 1 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Thanks @crenshaw-dev ! I looked through the docs with `make serve-docs-local` but obviously missed that indention. Should be fixed now. Let me know if you have any remarks or need further examples.

File: docs/operator-manual/user-management/microsoft.md [hunk 2]
```diff
@@ -92,107 +80,125 @@
...
-               p, role:org-admin, applications, *, */*, allow
-               p, role:org-admin, clusters, get, *, allow
-               p, role:org-admin, repositories, get, *, allow
-               p, role:org-admin, repositories, create, *, allow
-               p, role:org-admin, repositories, update, *, allow
-               p, role:org-admin, repositories, delete, *, allow
-               g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:org-admin
-               scopes: '[roles, email]'
+### Dedicated security group for administrators
+To map a special security group to the ArgoCD `admin` role, you can use the following `argocd-rbac-cm` ConfigMap:
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:159]: a/ArgoCD/Argo CD

File: docs/operator-manual/user-management/microsoft.md [hunk 3]
```diff
@@ -92,107 +80,125 @@
...
+  policy.csv: |
+    # either use the security group name
+    g, "argocd_admin_SG", role:admin
+    # or the group Object ID
+    g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:admin
+```
 
-## Azure AD App Registration Auth using Dex
+### Detailled permission
+This `argocd-rbac-cm` ConfigMap maps the Azure AD security group `argocd_org_admin_SG` to the restricted ArgoCD role `org-admin`:
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:172]: s/ArgoCD/ArgoCD

File: docs/operator-manual/user-management/microsoft.md [hunk 4]
```diff
@@ -92,107 +80,125 @@
...
+  policy.default: role:readonly
+  policy.csv: |
+    # either use the security group name
+    g, "argocd_admin_SG", role:admin
+    # or the group Object ID
+    g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:admin
+```
 
-## Azure AD App Registration Auth using Dex
+### Detailled permission
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:171]: s/Detailled/Detailed

File: docs/operator-manual/user-management/microsoft.md [hunk 5]
```diff
@@ -31,48 +32,35 @@
...
-               p, role:org-admin, repositories, update, *, allow
-               p, role:org-admin, repositories, delete, *, allow
-               g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:org-admin # (azure group assigned to role)
-
-## Azure AD App Registration Auth using OIDC
-### Configure a new Azure AD App registration
-#### Add a new Azure AD App registration
+#### Configure Argo to use the new Azure AD Enterprise App
+
+Edit `argocd-cm` ConfigMap and add the following `dex.config` to the data section, replacing the `caData`, `my-argo-cd-url` and `my-login-url` your values from the Azure AD App:
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:37]: Can you add the word "with" after "`my-login-url`"?

File: docs/operator-manual/user-management/microsoft.md [hunk 6]
```diff
@@ -92,107 +80,125 @@
...
+Then, add the `dex.config` to the `argocd-cm` ConfigMap:
 
-            ConfigMap -> argocd-rbac-cm
+```yaml
+data:
+  dex.config: |
+    connectors:
+    - type: microsoft
+      id: microsoft
+      name: Your Company GmbH
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:140]: Can we drop "GmbH"?

File: docs/operator-manual/user-management/microsoft.md [hunk 7]
```diff
@@ -92,107 +80,125 @@
...
-            policy.csv: |
-               p, role:org-admin, applications, *, */*, allow
-               p, role:org-admin, clusters, get, *, allow
-               p, role:org-admin, repositories, get, *, allow
-               p, role:org-admin, repositories, create, *, allow
-               p, role:org-admin, repositories, update, *, allow
-               p, role:org-admin, repositories, delete, *, allow
-               g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:org-admin
+## Configuring authorization through RBAC
+Edit the `argocd-rbac-cm` ConfigMap to assign permissions from the user's Azure AD security group memberships. After being logged in into ArgoCD, check [the user's group membership](log-in-to-argocd-ui-using-sso) to decide if you have to use the security group's `Object ID` or `name`.
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/microsoft.md:153]: s/ArgoCD/Argo CD

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Just a few more nitpicks. Otherwise LGTM. :-)

File: docs/operator-manual/user-management/microsoft.md (hunk 4 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Fixed.

File: docs/operator-manual/user-management/microsoft.md (hunk 6 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Done.

File: docs/operator-manual/user-management/microsoft.md (hunk 7 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Fixed (and other occurences without whitespace)

File: docs/operator-manual/user-management/microsoft.md (hunk 5 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Added.

File: docs/operator-manual/user-management/microsoft.md (hunk 3 above)
Comment (schakko) [reply docs/operator-manual/user-management/microsoft.md]: Fixed (and other occurences without whitespace)

**Comment by blakepettersson** [comment]
This is pretty old, and there have been updates to the Azure AD (or Entra ID?) since. If there is anything that's still relevant to add, could you open a new PR?

**Pull request closed without merging**